src/App.tsx

```diff
@@ -67,18 +57,14 @@ export const App: React.FC = () => {
                   </div>
                 )}
 
-                {author && loaded && !hasError && posts.length === 0 && (
+                {author && !isLoading && !error && posts.length === 0 && (
                   <div className="notification is-warning" data-cy="NoPostsYet">
                     No posts yet
                   </div>
                 )}
 
-                {author && loaded && !hasError && posts.length > 0 && (
-                  <PostsList
-                    posts={posts}
-                    selectedPostId={selectedPost?.id}
-                    onPostSelected={setSelectedPost}
-                  />
+                {author && !isLoading && !error && posts.length > 0 && (
+                  <PostsList />
```

Comment:
The conditional rendering of the 'PostsList' component is correct. Ensure that the 'PostsList' component is properly defined and imported.